object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/MustCallInvokedChecker.java

```diff
@@ -514,41 +636,57 @@ private static boolean isTryWithResourcesVariable(LocalVariableNode lhs) {
    * the check fails.
    */
   private void checkMustCall(
-      LocalVarWithTree localVarWithTree, CFStore store, String outOfScopeReason) {
-    List<String> mustCallValue = typeFactory.getMustCallValue(localVarWithTree.tree);
+      Set<Set<LocalVarWithTree>> defs,
+      Set<LocalVarWithTree> localVarWithTreeSet,
+      CFStore store,
+      String outOfScopeReason) {
+
+    List<String> mustCallValue =
+        typeFactory.getMustCallValue(localVarWithTreeSet.iterator().next().tree);
     // optimization: if there are no must-call methods, we do not need to perform the check
     if (mustCallValue.isEmpty()) {
       return;
     }
     AnnotationMirror cmAnno;
 
-    // sometimes the store is null!  this looks like a bug in checker dataflow.
-    // TODO track down and report the root-cause bug
-    CFValue lhsCFValue = store != null ? store.getValue(localVarWithTree.localVar) : null;
-    if (lhsCFValue != null) { // When store contains the lhs
-      cmAnno =
-          lhsCFValue.getAnnotations().stream()
-              .filter(anno -> AnnotationUtils.areSameByClass(anno, CalledMethods.class))
-              .findAny()
-              .orElse(typeFactory.top);
-    } else {
-      cmAnno =
-          typeFactory
-              .getAnnotatedType(localVarWithTree.localVar.getElement())
-              .getAnnotationInHierarchy(typeFactory.top);
-    }
-
-    if (!calledMethodsSatisfyMustCall(mustCallValue, cmAnno)) {
-      if (!reportedMustCallErrors.contains(localVarWithTree)) {
-        reportedMustCallErrors.add(localVarWithTree);
+    Set<LocalVarWithTree> defsCopy = new LinkedHashSet<>(localVarWithTreeSet);
```

Comment:
Why do we need the `defsCopy` and do `iter.remove()` at the end of the loop?  That looks unnecessary to me

Reply:
You're right, I'll remove it.

Reply:
Remove `defsCopy` also?